pkg/vm/builtins/bitwise.go

```diff
@@ -26,14 +31,14 @@ func (b *Bitwise) InferValue(segment *memory.Segment, offset uint64) error {
 	xOffset := offset - bitwiseIndex
 	yOffset := xOffset + 1
 
-	xValue, err := segment.Read(xOffset)
-	if err != nil {
-		return err
+	xValue := segment.Peek(xOffset)
```

Comment:
Why the change from `Read` to `Peek`?

Reply:
To give a more helpful error message, if not the error message looks like:
"bitwise: bitwise: error_msg"